plugins/aea-test-autonomy/aea_test_autonomy/base_test_classes/contracts.py

```diff
@@ -53,6 +54,7 @@ class BaseContractTest(ABC):
     identifier: str = EthereumCrypto.identifier
     deployer_crypto: Crypto
     contract_address: Optional[str] = None
+    deploy_contract: bool = True
```

Comment:
Deploying the contracts should be optional, the registries image comes with all of required contracts pre deployed

Reply:
why not set it to false by default then?

Reply:
Because I thought setting it false by default would introduce a breaking change since other contract require deployment. If that's not the case I can change this

Reply:
Okay I see what you mean, you could do something like this in the child class if you dont want to delpoy:
https://github.com/valory-xyz/autonomous-fund/blob/main/packages/balancer/agents/autonomous_fund/tests/test_contracts/test_managed_pool/test_contract.py#L74

Reply:
I tried and I get error

```
function_abi = {'inputs': [{'internalType': 'string', 'name': '_name', 'type': 'string'}, {'internalType': 'string', 'name': '_symbol...nternalType': 'string', 'name': '_baseURI', 'type': 'string'}], 'stateMutability': 'nonpayable', 'type': 'constructor'}
args = (), kwargs = {}

    def merge_args_and_kwargs(
        function_abi: ABIFunction, args: Sequence[Any], kwargs: Dict[str, Any]
    ) -> Tuple[Any, ...]:
        """
        Takes a list of positional args (``args``) and a dict of keyword args
        (``kwargs``) defining values to be passed to a call to the contract function
        described by ``function_abi``.  Checks to ensure that the correct number of
        args were given, no duplicate args were given, and no unknown args were
        given.  Returns a list of argument values aligned to the order of inputs
        defined in ``function_abi``.
        """
        # Ensure the function is being applied to the correct number of args
        if len(args) + len(kwargs) != len(function_abi.get('inputs', [])):
>           raise TypeError(
                "Incorrect argument count.  Expected '{0}'.  Got '{1}'".format(
                    len(function_abi['inputs']),
                    len(args) + len(kwargs),
                )
            )
E           TypeError: Incorrect argument count.  Expected '3'.  Got '0'

../.local/share/virtualenvs/open-autonomy-VW6Otnns/lib/python3.10/site-packages/web3/_utils/abi.py:431: TypeError
```